lib/rust/parser/src/syntax/expression/types.rs

```diff
@@ -19,7 +29,7 @@ use std::fmt::Debug;
 #[derive(Debug)]
 pub struct Operator<'s> {
```

Comment:
Please apologize such a _newbie question_: Why this is not `Operator<S>`, but rather `<'s`>? I am not able to find the explanation at [rust by example](https://doc.rust-lang.org/rust-by-example/).

Reply:
`S` would be the name of a type parameter, used to define a similar type of generic data to C++. 

A parameter starting with `'` is a lifetime parameter, used to track data ownership and borrowing. Within the parser, there's a convention of naming a lifetime `'s` if it borrows from the source code being parsed. So most of the AST types have a `'s` parameter (even if they aren't generic in the conventional sense), because they (transitively) contain tokens, which contain string slices borrowed from the input.